### PR TITLE
network,dhcp4: drop "duid-only" option support in ClientIdentifier=

### LINKED ISSUE #25562: Dhcp clientidentifier duid-only not working correctly
### systemd version the issue has been seen with

249

### Used distribution

Ubuntu 22.04

### Linux kernel version used

5.15

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

I have a dhcp server which expects a specific clientidentifier.
The following enhancement would apply for a correct way to specify a custom clientidentifier but that does not seem to work correctly: https://github.com/systemd/systemd/issues/7828.

It looks like it always prepends something before the DUIDRawData also when specifying the option : 
ClientIdentifier=duid-only

 

### Unexpected behaviour you saw

When i put in the dhclient.conf the following option:
`send dhcp-client-identifier 31:31:31:31;`
the output from dhcpdump is:
`OPTION:  61 (  4) Client-identifier         31:31:31:31`

When i set this in the networkconfig for my interface:
`[Match]`
`Name=enp6*`
`[Network]`
`DHCP=ipv4`
`[DHCPv4]`
`ClientIdentifier=duid-only`
`DUIDType=link-layer`
`DUIDRawData=31:31:31:31`

Then i get the following line in dhcpdump:

`OPTION:  61 (  7) Client-identifier         ff:00:00:00:00:00:03
`

So it seems that the option duid-only is not working or I have missed a configuration step.

### Steps to reproduce the problem

see above.

### Additional program output to the terminal or log subsystem illustrating the issue

_No response_

### PR DESCRIPTION
**Pull request by yuwata**
Closes #25562.
Replaces #28371.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -492,6 +492,10 @@ CHANGES WITH 254 in spe:
         * The DHCPv4 + DHCPv6 + IPv6 RA logic in networkd gained support for
           the RFC8910 captive portal option.
 
+        * The "duid-only" option for DHCPv4 client's ClientIdentifier= setting
+          is now dropped, as it has never been worked, hence should not be used
```
Comment (bluca) [inline NEWS]: "as it never worked, hence it should not be used"

File: NEWS [hunk 2]
```diff
@@ -7877,7 +7881,8 @@ CHANGES WITH 239:
           hibernates again.
 
         * networkd's ClientIdentifier= now accepts a new option "duid-only". If
-          set the client will only send a DUID as client identifier.
+          set the client will only send a DUID as client identifier. (EDIT: the
+          option has been completely broken, and is dropped in v254.)
```
Comment (bluca) [inline NEWS]: "the option was broken, and was dropped in v254"

**Review by bluca — APPROVED** [review]

**Comment by yuwata** [comment]
@bluca Thank you for review. The suggestions are applied. Upgrading the green label.

**Comment by yuwata** [comment]
As the green tag was changed to `after-next-release`, the NEWS entry is moved to v255.

**Comment by bluca** [comment]
> As the green tag was changed to `after-next-release`, the NEWS entry is moved to v255.

is that ok? it seems a bit late after rc2 to drop a feature, even if not working

**Comment by yuwata** [comment]
> > As the green tag was changed to `after-next-release`, the NEWS entry is moved to v255.
> 
> is that ok? it seems a bit late after rc2 to drop a feature, even if not working

No problem. It's ok to postpone.

**Comment by bluca** [comment]
@yuwata needs rebase

**Pull request merged**